### PR TITLE
(feat: Top-K) Add page_table_transform Output Fusion for GVR_V2

### PR DESCRIPTION
**Pull request by akhilg-nv**
## 📌 Description

Adds a GVR backend to the existing fused top-k/page-table API. After selecting the logical top-k indices, we translate them into physical kv cache slots within the same kernel. This eliminates a separate conversion kernel. 

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [ ] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [ ] I have installed the hooks with `pre-commit install`.
- [ ] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [ ] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

## 🔬 Experimental Track

- [ ] This PR is **experimental**: it adds or changes code under `flashinfer/experimental/` and/or an `@flashinfer_experimental_api`. Tracking issue: #
  - [ ] The tracking issue names an owner, the reason for the experimental path, and a graduation plan with a target release.
  - [ ] Core changes are limited to a thin entry point (signature, shared validation, feature-gate check, backend selection, handoff).
  - [ ] Tests live in `tests/experimental/` and were validated on the intended hardware; a runnable example is included.
  - [ ] Nothing is registered in `flashinfer/aot.py`, and no experimental backend is reachable from `backend="auto"` without `FLASHINFER_ALLOW_EXPERIMENTAL_AUTO_BACKENDS=1`. (Calling an `@flashinfer_experimental_api` or naming a backend explicitly is itself the opt-in and needs no environment variable.)
  - [ ] **Test scope declared below.** The experimental CI lane runs exactly these targets, so keep them as narrow as the change allows.

```experimental-tests
# One target per line: a directory or a file. (A pytest ::selector is not
# supported -- the sharding runner cannot consume one.) Must be under
# tests/experimental/ and must exist. Delete these comment lines and add yours, e.g.
#
#   tests/experimental/test_my_backend.py
#   tests/experimental/my_backend/
#
# Declaring the whole tree (tests/experimental/) is allowed but means every
# experimental PR pays for every other feature's tests, in every matrix cell.
```

## Summary by CodeRabbit

- **New Features**
  - Added an optional fused GVR backend for page-table top-k selection.
  - Supports physical page-index translation, optional raw-index output, short-row padding, and supported register-based GVR kernel routes.
  - Added validation for unsupported configurations and overlapping output buffers.
  - Default backend routing remains unchanged.

- **Documentation**
  - Documented the new backend, output behavior, supported configurations, limitations, and CUDA graph considerations.

- **Tests**
  - Added coverage for page remapping, padding, CUDA graph replay, edge cases, and unsupported options.

### CODE REVIEW

File: flashinfer/topk.py [hunk 1]
```diff
@@ -1089,6 +1065,7 @@ def top_k_page_table_transform(
     page_size: int = 1,
     out: Optional[torch.Tensor] = None,
     out_raw_indices: Optional[torch.Tensor] = None,
+    backend: str = "auto",
```
Comment (dhiraj113) [inline flashinfer/topk.py:1068]: Is hooking up GVR_v2 backend to this old API the right choice. I was going for adding page table support for the `top_k_varlen.py` API.
